### ISSUE TITLE
Centerface PTQ onnx to trt build engine failed

### ISSUE THREAD

**Issue by joytsay** [NVIDIA/TensorRT#1556]
## Description

The pretrained  [pth](https://drive.google.com/file/d/1DB67YGsQvqOlm9U0Co9puyrgGcpnhEva/view?usp=sharing) model is from this repo: 
https://github.com/chenjun2hao/CenterFace.pytorch
and I did PTQ to get the [onnx](https://drive.google.com/file/d/1IfMfrjO1f6o8DqmXd7OcQEP8Ef-Du1y-/view?usp=sharing) successfully
but when converting the PTQ onnx to trt with trtexec it fails:
```
trtexec --onnx=quant_centerface-calibrated_160.onnx --best --saveEngine=quant_centerface-calibrated_160.trt --dumpProfile --separateProfileRun --workspace=1000
...
[10/13/2021-02:12:15] [E] Error[10]: [optimizer.cpp::computeCosts::1981] Error Code 10: Internal Error (Could not find any implementation for node dla_up.up_0.up.0.weight + QuantizeLinear_693_quantize_scale_node + ConvTranspose_697 + BatchNormalization_698.)
[10/13/2021-02:12:15] [E] Error[2]: [builder.cpp::buildSerializedNetwork::561] Error Code 2: Internal Error (Assertion enginePtr != nullptr failed. )

```
looks like something to do with:
 ` dla_up.up_0.up.0.weight + QuantizeLinear_693_quantize_scale_node + ConvTranspose_697 + BatchNormalization_698`
any work around or fix known?
it seems similar to #1519 can @ttyio verify? 
also similar to  #1461 and #209 (i'm scrambling around issues to find a fix)
p.s. The normal centerface pth to onnx to trt succeeds. Only the PTQ version onnx to trt fails

## Environment

**TensorRT Version**:  8200 EA
**NVIDIA GPU**: NVIDIA GeForce RTX 2080 Ti
**NVIDIA Driver Version**:  470.63.01
**CUDA Version**: 11.4
**CUDNN Version**: 8204
**Operating System**: ubuntu 20.04
**Python Version (if applicable)**: 3.8.10
**Tensorflow Version (if applicable)**:  N/A
**PyTorch Version (if applicable)**: 1.10.0a0+3fd9dcf
**Baremetal or Container (if so, version)**:  nv-tensorrt-repo-ubuntu2004-cuda11.4-trt8.2.0.6-ea-20210922_1-1_amd64.deb

## Relevant Files

- Original pth: [centerface_model_best.pth](https://drive.google.com/file/d/1DB67YGsQvqOlm9U0Co9puyrgGcpnhEva/view?usp=sharing)

- PTQ onnx: [quant_centerface-calibrated_160.onnx](https://drive.google.com/file/d/1IfMfrjO1f6o8DqmXd7OcQEP8Ef-Du1y-/view?usp=sharing)

## Steps To Reproduce

- cmd:
` trtexec --onnx=quant_centerface-calibrated_160.onnx --best --saveEngine=quant_centerface-calibrated_160.trt --dumpProfile --separateProfileRun --workspace=1000`

- log:
````
root@ac4831ac49bd:/quant-dir/model# trtexec --onnx=quant_centerface-calibrated_160.onnx --best --saveEngine=quant_centerface-calibrated_160.trt --dumpProfile --separateProfileRun --workspace=1000
&&&& RUNNING TensorRT.trtexec [TensorRT v8200] # trtexec --onnx=quant_centerface-calibrated_160.onnx --best --saveEngine=quant_centerface-calibrated_160.trt --dumpProfile --separateProfileRun --workspace=1000
[10/13/2021-02:11:46] [I] === Model Options ===
[10/13/2021-02:11:46] [I] Format: ONNX
[10/13/2021-02:11:46] [I] Model: quant_centerface-calibrated_160.onnx
[10/13/2021-02:11:46] [I] Output:
[10/13/2021-02:11:46] [I] === Build Options ===
[10/13/2021-02:11:46] [I] Max batch: explicit batch
[10/13/2021-02:11:46] [I] Workspace: 1000 MiB
[10/13/2021-02:11:46] [I] minTiming: 1
[10/13/2021-02:11:46] [I] avgTiming: 8
[10/13/2021-02:11:46] [I] Precision: FP32+FP16+INT8
[10/13/2021-02:11:46] [I] Calibration: Dynamic
[10/13/2021-02:11:46] [I] Refit: Disabled
[10/13/2021-02:11:46] [I] Sparsity: Disabled
[10/13/2021-02:11:46] [I] Safe mode: Disabled
[10/13/2021-02:11:46] [I] Strict mode: Disabled
[10/13/2021-02:11:46] [I] Restricted mode: Disabled
[10/13/2021-02:11:46] [I] Save engine: quant_centerface-calibrated_160.trt
[10/13/2021-02:11:46] [I] Load engine:
[10/13/2021-02:11:46] [I] Profiling verbosity: 0
[10/13/2021-02:11:46] [I] Tactic sources: Using default tactic sources
[10/13/2021-02:11:46] [I] timingCacheMode: local
[10/13/2021-02:11:46] [I] timingCacheFile:
[10/13/2021-02:11:46] [I] Input(s)s format: fp32:CHW
[10/13/2021-02:11:46] [I] Output(s)s format: fp32:CHW
[10/13/2021-02:11:46] [I] Input build shapes: model
[10/13/2021-02:11:46] [I] Input calibration shapes: model
[10/13/2021-02:11:46] [I] === System Options ===
[10/13/2021-02:11:46] [I] Device: 0
[10/13/2021-02:11:46] [I] DLACore:
[10/13/2021-02:11:46] [I] Plugins:
[10/13/2021-02:11:46] [I] === Inference Options ===
[10/13/2021-02:11:46] [I] Batch: Explicit
[10/13/2021-02:11:46] [I] Input inference shapes: model
[10/13/2021-02:11:46] [I] Iterations: 10
[10/13/2021-02:11:46] [I] Duration: 3s (+ 200ms warm up)
[10/13/2021-02:11:46] [I] Sleep time: 0ms
[10/13/2021-02:11:46] [I] Streams: 1
[10/13/2021-02:11:46] [I] ExposeDMA: Disabled
[10/13/2021-02:11:46] [I] Data transfers: Enabled
[10/13/2021-02:11:46] [I] Spin-wait: Disabled
[10/13/2021-02:11:46] [I] Multithreading: Disabled
[10/13/2021-02:11:46] [I] CUDA Graph: Disabled
[10/13/2021-02:11:46] [I] Separate profiling: Enabled
[10/13/2021-02:11:46] [I] Time Deserialize: Disabled
[10/13/2021-02:11:46] [I] Time Refit: Disabled
[10/13/2021-02:11:46] [I] Skip inference: Disabled
[10/13/2021-02:11:46] [I] Inputs:
[10/13/2021-02:11:46] [I] === Reporting Options ===
[10/13/2021-02:11:46] [I] Verbose: Disabled
[10/13/2021-02:11:46] [I] Averages: 10 inferences
[10/13/2021-02:11:46] [I] Percentile: 99
[10/13/2021-02:11:46] [I] Dump refittable layers:Disabled
[10/13/2021-02:11:46] [I] Dump output: Disabled
[10/13/2021-02:11:46] [I] Profile: Enabled
[10/13/2021-02:11:46] [I] Export timing to JSON file:
[10/13/2021-02:11:46] [I] Export output to JSON file:
[10/13/2021-02:11:46] [I] Export profile to JSON file:
[10/13/2021-02:11:46] [I]
[10/13/2021-02:11:47] [I] === Device Information ===
[10/13/2021-02:11:47] [I] Selected Device: NVIDIA GeForce RTX 2080 Ti
[10/13/2021-02:11:47] [I] Compute Capability: 7.5
[10/13/2021-02:11:47] [I] SMs: 68
[10/13/2021-02:11:47] [I] Compute Clock Rate: 1.545 GHz
[10/13/2021-02:11:47] [I] Device Global Memory: 11018 MiB
[10/13/2021-02:11:47] [I] Shared Memory per SM: 64 KiB
[10/13/2021-02:11:47] [I] Memory Bus Width: 352 bits (ECC disabled)
[10/13/2021-02:11:47] [I] Memory Clock Rate: 7 GHz
[10/13/2021-02:11:47] [I]
[10/13/2021-02:11:47] [I] TensorRT version: 8200
[10/13/2021-02:11:47] [I] [TRT] [MemUsageChange] Init CUDA: CPU +320, GPU +0, now: CPU 332, GPU 312 (MiB)
[10/13/2021-02:11:47] [I] Start parsing network model
[10/13/2021-02:11:47] [I] [TRT] ----------------------------------------------------------------
[10/13/2021-02:11:47] [I] [TRT] Input filename:   quant_centerface-calibrated_160.onnx
[10/13/2021-02:11:47] [I] [TRT] ONNX IR version:  0.0.6
[10/13/2021-02:11:47] [I] [TRT] Opset version:    13
[10/13/2021-02:11:47] [I] [TRT] Producer name:    pytorch
[10/13/2021-02:11:47] [I] [TRT] Producer version: 1.10
[10/13/2021-02:11:47] [I] [TRT] Domain:
[10/13/2021-02:11:47] [I] [TRT] Model version:    0
[10/13/2021-02:11:47] [I] [TRT] Doc string:
[10/13/2021-02:11:47] [I] [TRT] ----------------------------------------------------------------
[10/13/2021-02:11:48] [I] Finish parsing network model
[10/13/2021-02:11:48] [I] [TRT] [MemUsageSnapshot] Builder begin: CPU 459 MiB, GPU 348 MiB
[10/13/2021-02:11:48] [W] [TRT] Calibrator won't be used in explicit precision mode. Use quantization aware training to generate network with Quantize/Dequantize nodes.
[10/13/2021-02:11:50] [I] [TRT] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +496, GPU +214, now: CPU 963, GPU 562 (MiB)
[10/13/2021-02:11:50] [I] [TRT] [MemUsageChange] Init cuDNN: CPU +169, GPU +204, now: CPU 1132, GPU 766 (MiB)
[10/13/2021-02:11:50] [I] [TRT] Local timing cache in use. Profiling results in this builder pass will not be stored.
[10/13/2021-02:12:15] [W] [TRT] Some weights are outside of int8_t range and will be clipped to int8_t range.
[10/13/2021-02:12:15] [W] [TRT] Some weights are outside of int8_t range and will be clipped to int8_t range.
[10/13/2021-02:12:15] [W] [TRT] Some weights are outside of int8_t range and will be clipped to int8_t range.
[10/13/2021-02:12:15] [W] [TRT] Some weights are outside of int8_t range and will be clipped to int8_t range.
[10/13/2021-02:12:15] [W] [TRT] Some weights are outside of int8_t range and will be clipped to int8_t range.
[10/13/2021-02:12:15] [W] [TRT] Some weights are outside of int8_t range and will be clipped to int8_t range.
[10/13/2021-02:12:15] [E] Error[10]: [optimizer.cpp::computeCosts::1981] Error Code 10: Internal Error (Could not find any implementation for node dla_up.up_0.up.0.weight + QuantizeLinear_693_quantize_scale_node + ConvTranspose_697 + BatchNormalization_698.)
[10/13/2021-02:12:15] [E] Error[2]: [builder.cpp::buildSerializedNetwork::561] Error Code 2: Internal Error (Assertion enginePtr != nullptr failed. )
[10/13/2021-02:12:15] [E] Engine could not be created from network
[10/13/2021-02:12:15] [E] Building engine failed
[10/13/2021-02:12:15] [E] Failed to create engine from model.
[10/13/2021-02:12:15] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8200] # trtexec --onnx=quant_centerface-calibrated_160.onnx --best --saveEngine=quant_centerface-calibrated_160.trt --dumpProfile --separateProfileRun --workspace=1000


````

**Comment by spivakoa** [comment]
Seems like the same error I met using depth convolution followed by BatchNormalization. Current constraint for the number of output channels divisible by 4. I believe that it becomes same for the number of input channels in ConvTranspose layer.

**Comment by joytsay** [comment]
@spivakoa Did you alter the model layer somehow to workaround this constraint? If so, can you show me how you done it? Any tips would be helpful!

**Comment by joytsay** [comment]
@ttyio I'm a little confused. Do I skip DepthConv in the PTQ step (i.e. when I produce the quantized onnx) or is it when I use trtexec to do onnx2trt?

**Comment by oxana-nvidia** [comment]
Issue is still reproducible with TensorRT 8.4.  

I've filed a request for TensorRT developers to investigate
internal bug number 3659918

**Comment by oxana-nvidia** [comment]
The issue is fixed in TensorRT 8.5 release

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!
